docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,361 @@
+  trusted by browsers and clients such as ``curl``. This is the easiest type to
+  use because you do not need to configure clients. Obtain a certificate of this
+  type from:
+
+  *  A commercial certificate vendor
+  *  Your cloud infrastructure provider
+  *  A domain name registrar, such as Verisign or GoDaddy
+  *  A free certificate generator, such as
+     `letsencrypt.org <https://letsencrypt.org/>`_ or
+     `sslforfree.com <https://www.sslforfree.com/>`_
+
+* **Corporate trusted certificates** — A certificate trusted by browsers and
+  clients in your organization. Typically, a site's IT department runs a local
+  :ref:`certificate authority <glossCA>` and preconfigures clients and servers
+  to trust this CA.
+
+* **Generated self-signed certificates** — A certificate generated just for
+  Trino that is not automatically trusted by any client. Before using, make sure
+  you understand the :ref:`limitations of self-signed certificates
+  <self_signed_limits>`.
+
+The most convenient option is to purchase a PEM certificate from a Certificate
+Authority that is already listed in the Java truststore. This case requires no
+extra configuration for clients to *just work* when connecting to your server.
+
+Keys, certificates, and containers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Trino recognizes public and private :ref:`keys <glossKey>` as specified in the
+PKCS #1 standard, and recognizes X.509 :ref:`certificates <glossCert>`. Trino
+supports keys and certificates referenced directly as PEM-encoded files without
+a container, which is the preferred method and easiest to work with.
+
+Trino also supports keys and certificates imported into legacy
+:ref:`JKS <glossJKS>` keystore containers, as well as those imported into the
+binary :ref:`PKCS #12 <glossPKCS12>` archive file format.
+
+Make sure you obtain a certificate that is validated by a recognized
+:ref:`certificate authority <glossCA>`.
+
+Inspect received certificates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Be sure to inspect and validate your received key and certificate files to make
+sure they reference the correct information to access your Trino server. Much
+unnecessary debugging time is saved by taking the time to validate your
+certificates before proceeding to configure the server.
+
+Inspect PEM-encoded files as described in :doc:`Inpsect PEM files
+</security/inspect-pem>`.
+
+Inspect JKS keystores as described in :doc:`Inspect JKS files
+</security/inspect-jks>`.
+
+Invalid certificates
+^^^^^^^^^^^^^^^^^^^^^
+
+If your certificate does not pass validation, or does not show the expected
+information on inspection, contact the group or vendor who provided it for a
+replacement.
+
+.. _cert-placement:
+
+Place the certificate file
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are no location requirements for a certificate file as long as:
+
+* The location is visible from the server's configuration file location with a
```

Comment:
I don't think this point is important, but it is important the file can be read by the Trino server process.

Reply:
OK